homeassistant/components/esphome/dashboard.py

```diff
@@ -21,23 +24,40 @@ def async_get_dashboard(hass: HomeAssistant) -> ESPHomeDashboard | None:
     return hass.data.get(KEY_DASHBOARD)
 
 
-def async_set_dashboard_info(
+async def async_set_dashboard_info(
     hass: HomeAssistant, addon_slug: str, host: str, port: int
 ) -> None:
     """Set the dashboard info."""
-    hass.data[KEY_DASHBOARD] = ESPHomeDashboard(
-        hass,
-        addon_slug,
-        f"http://{host}:{port}",
-        async_get_clientsession(hass),
-    )
+    url = f"http://{host}:{port}"
+
+    # Do nothing if we already have this data.
+    if (
+        (cur_dashboard := hass.data.get(KEY_DASHBOARD))
+        and cur_dashboard.addon_slug == addon_slug
+        and cur_dashboard.url == url
+    ):
+        return
+
+    dashboard = ESPHomeDashboard(hass, addon_slug, url, async_get_clientsession(hass))
+    try:
+        await dashboard.async_request_refresh()
+    except UpdateFailed:
+        return
```

Comment:
Isn't it odd to ignore this?
As in, it will be hard to debug if it doesn't work, right?

Reply:
I don't ignore it, I will not store the dashboard information. The coordinator should have already logged an error.